src/Microsoft.Health.Dicom.Core/Features/CustomTag/DeleteCustomTagService.cs

```diff
@@ -43,9 +45,9 @@ public async Task DeleteCustomTagAsync(string tagPath, CancellationToken cancell
 
             string normalizedPath = tags[0].GetPath();
 
-            CustomTagEntry customTagEntry = await _customTagStore.GetCustomTagAsync(normalizedPath, cancellationToken);
+            IEnumerable<CustomTagEntry> customTagEntries = await _customTagStore.GetCustomTagsAsync(normalizedPath, cancellationToken);
```

Comment:
We do currently chose to explicitly return IEnumerable in most of our get calls from store! Do you have concerns with us using Linq to get First?

---
In reply to: [577190537](https://github.com/microsoft/dicom-server/pull/542#discussion_r577190537) [](ancestors = 577190537)

Reply:
It would be faster and more efficient to avoid Linq as much as possible lol, especially when the underlying type is something like a `List<T>` which can quickly fetch the first element. Instead, `First()` has to allocate the `IEnumerator<T>` and call multiple members internally like `MoveNext()` and `Current`. Will this cause noticeable issues now? No haha. But these allocations do add up over time, and if we can nip these now it'd be preferrable.